python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -1079,6 +1153,134 @@ def _get_task_parent_entity_type(self, task_id):
 
         return cache[self.TASK_PARENT_TYPES][task_id]
 
+    def _legacy_get_project_actions(self, config_paths):
+        """
+        Gets all actions for all shotgun_xxx environments for the project and
+        caches them in memory, keyed by the unique session key provided by
+        the websocket server.
+
+        :param list config_paths: A list of string file paths to the root
+            directory of each pipeline configuration to get actions for.
+
+        :returns: All commands for all shotgun_xxx environments for all
+            requested pipeline configs.
+        :rtype: dict
+        """
+        # The in-memory cache is keyed by the wss_key that is unique to each
+        # wss connection.
+        if self.LEGACY_PROJECT_ACTIONS not in self._cache:
+            self._cache[self.LEGACY_PROJECT_ACTIONS] = self.process_manager.get_project_actions(
+                config_paths,
+            )
+
+        # We'll deepcopy the data before returning it. That will ensure that
+        # any destructive operations on the contents won't bubble up to the
+        # cache.
+        return copy.deepcopy(self._cache[self.LEGACY_PROJECT_ACTIONS])
+
+    def _legacy_process_configs(self, config_data, entity_type):
+        """
+        Processes the raw engine command data coming from the tank command
+        and organizes it into the data structure expected from the v2 wss
+        server by the client. This method acts as the adapter between the
+        legacy v1 API method of getting toolkit actions and the v2 menu
+        logic in the Shotgun web app versions 7.2.0+.
+
+        :param dict config_data: A dictionary, keyed by PipelineConfiguration
+            entity name, containing a tuple of config root path and config
+            entity dict, in that order.
+        :param str entity_type: The entity type that we're getting actions
+            for.
+        """
+        config_paths = [p[0] for n, p in config_data.iteritems()]
+        project_actions = self._legacy_get_project_actions(config_paths)
+        all_actions = dict()
+        config_names = []
+
+        for config_name, config_data in config_data.iteritems():
+            config_path, config_entity = config_data
+            commands = []
+            get_actions_data = project_actions[config_path]["shotgun_get_actions"]
+            env_file_name = "shotgun_%s.yml" % entity_type.lower()
+            raw_actions_data = get_actions_data[env_file_name]
```

Comment:
NOTE: This isn't correct. It needs to be a .get and then report an unsupported entity type to the client if None. I'll fix it Monday.